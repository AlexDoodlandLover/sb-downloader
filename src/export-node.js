import environment from './environment';
environment.canAccessScratchAPI = true;

export * from './downloader';
