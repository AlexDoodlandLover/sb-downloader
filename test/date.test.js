import {expect, test} from 'vitest';
import fs from 'fs';
import JSZip from 'jszip';
import {getFixturePath, arrayBufferSerializer} from './test-utilities.js';
import * as SBDL from '../src/export-node.js';

expect.addSnapshotSerializer(arrayBufferSerializer);

test('date defaults to a constant arbitrary time', async () => {
  const project = await SBDL.downloadProjectFromBuffer(fs.readFileSync(getFixturePath('minimal-sb3.json')));
  const zip = await JSZip.loadAsync(project.arrayBuffer);
  expect(zip.files['project.json'].date.toUTCString()).toBe('Fri, 31 Dec 2021 00:00:00 GMT');
  expect(zip.files['592bae6f8bb9c8d88401b54ac431f7b6.svg'].date.toUTCString()).toBe('Fri, 31 Dec 2021 00:00:00 GMT');
  expect(zip.files['83a9787d4cb6f3b7632b4ddfebf74367.wav'].date.toUTCString()).toBe('Fri, 31 Dec 2021 00:00:00 GMT');
  expect(project).toMatchSnapshot();
}, 30000);

test('date option', async () => {
  const date = new Date(1000030000000);
  const project = await SBDL.downloadProjectFromBuffer(fs.readFileSync(getFixturePath('minimal-sb3.json')), {
    date
  });
  const zip = await JSZip.loadAsync(project.arrayBuffer);
  expect(zip.files['project.json'].date.toUTCString()).toBe(date.toUTCString());
  expect(zip.files['592bae6f8bb9c8d88401b54ac431f7b6.svg'].date.toUTCString()).toBe(date.toUTCString());
  expect(zip.files['83a9787d4cb6f3b7632b4ddfebf74367.wav'].date.toUTCString()).toBe(date.toUTCString());
  expect(project).toMatchSnapshot();
}, 30000);
