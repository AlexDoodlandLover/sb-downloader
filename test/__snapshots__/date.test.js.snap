// Vitest Snapshot v1

exports[`date defaults to a constant arbitrary time 1`] = `
{
  "arrayBuffer": ArrayBuffer [SHA-256 36e8b445404a8072426a582dbc689028e49e391d301860ecc1df92ab98dbf527],
  "title": "",
  "type": "sb3",
}
`;

exports[`date option 1`] = `
{
  "arrayBuffer": ArrayBuffer [SHA-256 ba0c40627982515f228f62dbe8b18fbad476f7e6294d78b7a794f354f42afa0f],
  "title": "",
  "type": "sb3",
}
`;
